dataprofiler/data_readers/data_utils.py

```diff
@@ -243,13 +271,13 @@ def read_json(data_generator, selected_columns=None, read_in_string=False):
 
 
 def read_csv_df(
-    file_path,
-    delimiter,
-    header,
-    selected_columns=[],
-    read_in_string=False,
-    encoding="utf-8",
-):
+    file_path: Union[str, BytesIO, TextIOWrapper],
```

Comment:
similar to below, if we override allowing each diff input, does that remove the need to cast?

Reply:
I don't think so because as far I can tell, overloading doesn't let us have multiple implementations. It just lets us specify relationships between input types and return types.

Reply:
in certain conditions, would still need to cast I think because the code doesn't know that under this condition the type should be this. So that's where `cast()` comes in to play under specific `if` conditions... at least that's how I'm reading it